clients/go/pkg/worker/jobPoller_test.go

```diff
@@ -204,6 +204,30 @@ func (suite *JobPollerSuite) TestShouldIgnoreStreamFailure() {
 	suite.consumeJob()
 }
 
+func (suite *JobPollerSuite) TestShouldReportMetrics() {
```

Comment:
Is this testing that the metric is reset to 0 on close?

Reply:
Yes, the test includes also resetting the metric to 0 on close.